core/ArchiveProcessor/PluginsArchiver.php

```diff
@@ -60,7 +66,24 @@ public function __construct(Parameters $params, $isTemporaryArchive, ArchiveWrit
 
         $this->archiveProcessor = new ArchiveProcessor($this->params, $this->archiveWriter, $this->logAggregator);
 
-        $this->isSingleSiteDayArchive = $this->params->isSingleSiteDayArchive();
+        $shouldAggregateFromRawData = $this->params->isSingleSiteDayArchive();
+
+        /**
+         * Triggered to detect if the archiver should aggregate from raw data by using MySQL queries (when true)
+         * or by aggregate archives (when false). Typically, data is aggregated from raw data for "day" period, and
+         * aggregregated from archives for all other periods.
+         *
+         * @param bool $shouldAggregateFromRawData  Set to true, to aggregate from raw data, or false to aggregate multiple reports.
+         * @param Parameters $params
+         * @ignore
+         * @deprecated
+         *
+         * In Matomo 4.0 we should maybe remove this event, and instead maybe always archive from raw data when it is daily archive,
+         * no matter if single site or not. We cannot do this in Matomo 3.X as some custom plugin archivers may not be able to handle multiple sites.
+         */
+        Piwik::postEvent('ArchiveProcessor.shouldAggregateFromRawData', array(&$shouldAggregateFromRawData, $this->params));
```

Comment:
I've set the event to be ignored for now. I can see it may be useful if someone actually wanted to aggregate week or month from raw data as well but usually that doesn't happen. It will be useful though in the meantime not only for the tests but also for Roll-Up reporting which soon may or may not force the archiving from raw data when there are multiple sites.